src/operator.py

```diff
@@ -0,0 +1,375 @@
+# Interface -> Server Handlers #
+################################
+@custom_service
+def all_drones_available(request, response):
+    drones_available = []
+    print("Calling all_drones_available service...")
+    for k, v in drones.items():
+        avail = dict()
+        avail["id"] = k
+        avail["name"] = v.drone_name
+        avail["type"] = v.drone_type
+        avail["topics"] = v.topics
+        avail["services"] = v.services
+        drones_available.append(avail)
+
+    response["success"] = True
+    response["message"] = "Successfully sent all available drones to VR"
+    response['drones_available'] = drones_available
+    print("All_drones_available service finished!")
+    return True
+
+
+@custom_service
+def upload_mission(request, response):
+    '''
+    :param request: dict of {drone_id: int, waypoints: list of ints/strings --> pass
+    these directly into the drone instance}
+    '''
+    print("Calling upload_mission service...")
+    if not drones.get(request["drone_id"]):
+        response["success"] = False
+        response["message"] = "No drone with that id."
+        response["drone_id"] = -1
+        return False
+    response = drones[request["drone_id"]].upload_mission(request["waypoints"])
+    print("Upload_mission service finished!")
+    return True
+
+
+@custom_service
+def set_speed(request, response):
+    print("Calling set_speed service...")
+    if not drones.get(request["drone_id"]):
+        response["success"] = False
+        response["message"] = "No drone with that id."
+        response["drone_id"] = -1
+        return False
+    print('Setting speed to {}...'.format(request['data']))
+    response = drones[request["drone_id"]].set_speed(request["data"])
+    response['success'] = True
+    print("Set_speed service finished!")
+    return True
+
+
+# includes startmission, pausemission, resume mission, landdrone, flyhome
+@custom_service
+def control_drone(request, response):
+    print("Calling control_drone service...")
+    control_task = request["control_task"]
+    drone = drones.get(request["id"])
+    if not drone:
+        response["success"] = False
+        response["message"] = "Invalid drone id"
+        return False
+    if control_task == "start_mission":
+        print("Executing start_mission...")
+        response = drone.start_mission()
+    elif control_task == "pause_mission":
+        print("Executing pause_mission...")
+        response = drone.pause_mission()
+    elif control_task == "resume_mission":
+        print("Executing resume_mission...")
+        response = drone.resume_mission()
+    elif control_task == "land_drone":
+        print("Executing land_drone...")
+        response = drone.land_drone()
+    elif control_task == "fly_home":
+        print("Executing fly_home...")
+        response = drone.fly_home()
+    else:
+        response["success"] = False
+        response["message"] = "Invalid control task"
+        return False
```

Comment:
Doesn't change functionality, but does use fewer repeated print statements:

```suggestion
    tasks = {
        "start_mission" : drone.start_mission,
        "pause_mission" : drone.pause_mission,
        "resume_mission" : drone.resume_mission,
        "land_drone" : drone.land_drone,
        "fly_home" : drone.fly_home
    }
    if control_task not in tasks:
        response["success"] = False
        response["message"] = "Invalid control task"
        return False
    else:
        print(f"Executing {control_task}...")
        response = tasks.get(control_task)()
```